### PR TITLE
Update test suite.

### PR DESCRIPTION
**Pull request by ghost**
Update many tests to use cleaner TEST_ syntax.

### CODE REVIEW

**Comment by stepancheg** [comment]
Changes that can be split into separate commit should be separate commits.

File: lib-clay/core/operators/operators.clay [hunk 1]
```diff
@@ -111,7 +111,6 @@ forceinline overload assign(ref dest:T, ref src:T) : {
 [T when
     not BitwiseAssignedType?(T)
     and CallDefined?(copy, T)
-    and CallDefined?(move, T)
```
Comment (stepancheg) [inline lib-clay/core/operators/operators.clay]: This change is not described in commit message (it should be, in fact, separate commit).

File: lib-clay/test/test.clay [hunk 2]
```diff
@@ -167,7 +159,7 @@ overload expect(passed?: Bool, failReportMsg:String) {
     }
 }
 
-alias overload expect(passed?:Bool) {
```
Comment (stepancheg) [inline lib-clay/test/test.clay]: This change is not in commit message either.

**Comment by stepancheg** [comment]
What's wrong with Function in TestCase? TestCase is generic now, so I can't (for instance) just declare a variable of type Maybe[TestCase].

**Comment by jckarter** [comment]
Updating tests to use test.module is nice, but I would keep TestCase using arrays and Function. As @stepancheg noted, it's nice to be able to refer to test cases as values with concrete type, and it also eliminates a bunch of instantiation.

File: lib-clay/core/operators/operators.clay (hunk 1 above)
Comment (ghost) [inline lib-clay/core/operators/operators.clay]: http://www.buyartspaintings.com burberry pas cher
[url=http://www.mbtshoesmbts.com]cheap mbt shoes[/url]

File: lib-clay/test/test.clay (hunk 2 above)
Comment (ghost) [inline lib-clay/test/test.clay]: <a href="http://www.minilicores.com">lebron 11 shoes</a> U 8 <a href="http://www.zeeshan.im">lebron 11 for sale</a> O 3  <a href="http://www.williamspayne.com">nike lebron 11</a> R 7  <a href="http://www.academicflair.com">Lebron 11 Gold</a> Y 8  <a href="http://www.sinautecus.com">nike lebron 11</a> F 7  <a href="http://lamlicious.com">nike lebron 11</a> Q 6  <a href="http://www.delta-research.net">cheap lebron 11</a> D 3  <a href="http://www.force-fed-fabrications.com">lebron 11 for sale</a> S 3  <a href="http://www.bigpalooka.com">lebron 11</a> T 3  <a href="http://www.aromesgroupe.com">lebron 11 shoes</a> Y 4  <a href="http://www.thecompdr.net">Jordan 6 Bright Grape</a> S 8  <a href="http://www.wspra.net">Jordan 6 Bright Grape</a> R 5  <a href="http://www.parkerfranklin.com">Bright Grape 6s</a> O 3  <a href="http://www.dlmachine.com">Bright Grape 6</a> T 5 <a href="http://www.bi-ritealum.com">Jordan 6 Bright Grape</a> U 3 <a href="http://www.springcreekhomes.org">Jordan 6 Bright Grape</a> S 6 <a href="http://www.litovkin.com">Jordan 6 Bright Grape</a> D 5 <a href="http://www.kingdavidtours.org">Retro 6 Bright Grape</a> Q 7 <a href="http://www.jeikor.com">Bright Grape 6s</a> W 5 <a href="http://www.wrmediadesign.com">Jordan Retro 6</a> Y 7
louis vuitton bags http://www.mktravelclinic.com/

**Pull request merged**